### PR TITLE
Handle exceptions in main

### LINKED ISSUE #12: Raise exceptions and handle them in main function
Base on @JohanLorenzo 's great idea [1], we can just raise exception when there is an error and handle them in main function.

[1] https://github.com/askeing/taskcluster-util-python/pull/6#issuecomment-126220189

### PR DESCRIPTION
**Pull request by askeing**
fix #12

### CODE REVIEW

**Comment by askeing** [comment]
@JohanLorenzo r?

File: taskcluster_util/taskcluster_download.py [hunk 1]
```diff
@@ -65,8 +65,7 @@ def run(self):
         # check credentials file
         abs_credentials_path = os.path.abspath(self.options.credentials)
         if not os.path.isfile(abs_credentials_path):
-            logger.warning('{} is not a file or is not exist.'.format(abs_credentials_path))
-            exit(-1)
+            raise Exception('{} is not a file or is not exist.'.format(abs_credentials_path))
```
Comment (JohanLorenzo) [inline taskcluster_util/taskcluster_download.py]: After checking, this exception is already given for free by the `with` in `Credentials.from_file()`. If I delete this line and the one just above, I can get:

`2015-07-31 13:46:24,339 - __main__ 103 - ERROR - [Errno 21] Is a directory: '/home/jlorenzo/git/taskcluster-util-python/test-dir'`
`2015-07-31 13:44:38,635 - __main__ 105 - ERROR - [Errno 2] No such file or directory: '/home/jlorenzo/git/taskcluster-util-python/credential_non_existing.json'`

File: taskcluster_util/taskcluster_download.py [hunk 2]
```diff
@@ -87,9 +86,8 @@ def run(self):
                 task_id = task_finder.get_taskid_by_namespace(task_namespace)
                 logger.info('The TaskID of Namespace [{}] is [{}].'.format(task_namespace, task_id))
             except TaskclusterRestFailure as e:
-                logger.warning('Can not get the TaskID due to [{}]'.format(e.message))
                 logger.error(e.body)
```
Comment (JohanLorenzo) [inline taskcluster_util/taskcluster_download.py]: After taking a closer look, `e.body` if defined in TaskCluster errors only. I understand why we log the values here. I'm thinking of a way to simplify `run()` a little bit more. What do you think, if we remove every try/except in `run()`, and change main() this way?

``` python
def main():
    try:
        DownloadRunner().run()
    except Exception as e:
        logger.error(e)
        if e.__dict__:
            logger.error(e.__dict__)
        exit(1)
```

Printing the `__dict__` will enable us to output all the details of the exception, how ever the exception is made. And so we could also clean the try/excepts left in `downloader.py`. What do you think?

**Comment by askeing** [comment]
after do the modification, the run() become more clear :)
@JohanLorenzo r?

**Comment by JohanLorenzo** [comment]
It looks great! r+
Thanks a lot!

**Comment by askeing** [comment]
merged
@JohanLorenzo thanks for your suggestions :)

**Pull request merged**